### ISSUE TITLE
Applying gradientBlur causes unintended blurring

### ISSUE THREAD

**Issue by bbsody** [Eskils/VariableBlurImageView#2]
The `gradientBlur` method is currently causing unintended blurring in areas where the provided gradient image has zero brightness.

**Steps to Reproduce:**
1. Use the gradientBlur method (or the equal `variableBlurEngine` method and SwiftUI modifier) with a gradient image where some areas have zero brightness.
2. Observe the blurring effect in the corresponding areas of the output.

**Expected Behavior:**
The gradientBlur method should apply the blurring effect based on the gradient image, considering the brightness values. Areas with zero brightness in the gradient image should not contribute to blurring.

**Actual Behavior:**
The blurring is applied in areas where the gradient image has zero brightness, leading to an undesirable outcome.

**Comment by Eskils** [comment]
Hey @bbsody,

Thank you for your descriptive feedback. It’s greatly appreciated!

I did some testing and was unable to reproduce your scenario. Are you able to provide an example gradient image which causes the unexpected behaviour?

Thank you in advance. :’)

Best,
Eskil

---

The code I used:

```swift
try variableBlurEngine.applyGradientVariableBlur(
    toImage: input,
    withGradient: gradientImage,
    maxRadius: 20
)
```

| Gradient Image | Result Image |
|----------------|---------------|
| ![Gradient Image](https://github.com/Eskils/VariableBlurImageView/assets/26850613/46930d23-65fd-4977-ab3e-b72180609e8b) | ![Result Image](https://github.com/Eskils/VariableBlurImageView/assets/26850613/9ac3d9a9-d17a-4ad3-90c6-9bc7b45d4968) |

**Comment by bbsody** [comment]
@Eskils, thank you for a quick reply (and a great project)!

I first noticed the lack of sharpness of the image in your demo project without making any alterations on my end. Then, I tried to use another gradient image with a more distinct black area but got the same result.

![gradient_example](https://github.com/Eskils/VariableBlurImageView/assets/694273/1dceb5a7-7478-4493-bed0-5d38d6d7bb6e)
![gradientImage](https://github.com/Eskils/VariableBlurImageView/assets/694273/3a7061fd-914c-46d2-b0bb-13b2129003ad)

I haven't had the opportunity to check the rendered image on a non-Retina screen yet, but I don't expect a difference in the results.

**Comment by Eskils** [comment]
Ah, I understand. 

It appears that the unintended blur is more prominent on smaller images. I’ll start to work on a potential bug fix.

Thank you for all the information you have provided. 🙌

**Comment by Eskils** [comment]
The fix for this issue has been merged and is implemented in [VariableBlurImageView 1.1.2](https://github.com/Eskils/VariableBlurImageView/releases/tag/v1.1.2). 

Thank you for reporting and for your helpful feedback! 🚀

**Comment by bbsody** [comment]
Thank you so much for addressing the issue promptly!
